### PR TITLE
Refactor Code samples; Test code samples

### PR DESCRIPTION
**Pull request by LysandreJik**
Refactoring the code samples in order to prevent copy/pasting the same code samples across classes while updating the model/tokenizer classes and checkpoint names.

- All models now have their docstrings updated. 
- Doctest is used for testing
- Fixed a bunch of bugs in all docstrings as well as a few models. All non-cosmetic changes are highlighted below.

### CODE REVIEW

**Comment by sgugger** [comment]
This is amazing! This way we won't do as many mistakes while copy-pasting code for introducing those task-specific models :-)

File: src/transformers/file_utils.py [hunk 1]
```diff
@@ -149,6 +149,130 @@ def docstring_decorator(fn):
...
+        inputs = tokenizer("Hello, my dog is cute", return_tensors="pt")
+        outputs = model(**inputs)
+
+        last_hidden_states = outputs[0]  # The last hidden-state is the first element of the output tuple
+"""
+
+PYTORCH_MULTIPLE_CHOICE_CODE_SAMPLE_DOCSTRING = r"""
+    Examples::
+
+        from transformers import BertTokenizer, BertForMultipleChoice
```
Comment (thomwolf) [inline src/transformers/file_utils.py:233]: This should be « {tokenizer_class}, {model_class} «

File: src/transformers/file_utils.py [hunk 2]
```diff
@@ -149,6 +149,130 @@ def docstring_decorator(fn):
...
+        import torch
+
+        tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
```
Comment (thomwolf) [inline src/transformers/file_utils.py:236]: Change also here and below

File: src/transformers/file_utils.py [hunk 3]
```diff
@@ -149,6 +149,130 @@ def docstring_decorator(fn):
...
+        model = BertForMultipleChoice.from_pretrained('bert-base-uncased')
+
+        prompt = "In Italy, pizza served in formal settings, such as at a restaurant, is presented unsliced."
+        choice0 = "It is eaten with a fork and a knife."
+        choice1 = "It is eaten while held in the hand."
+        labels = torch.tensor(0).unsqueeze(0)  # choice0 is correct (according to Wikipedia ;)), batch size 1
+
+        encoding = tokenizer.batch_encode_plus([[prompt, choice0], [prompt, choice1]], return_tensors='pt', pad_to_max_length=True)
```
Comment (thomwolf) [inline src/transformers/file_utils.py:244]: And we probably want to remove batch_encode_plus here hahah

**Review by thomwolf — APPROVED** [review]
Yes I like this idea a lot! Will be easier to add more Tensorflow examples as well!

**Review by julien-c — APPROVED** [review]

File: src/transformers/modeling_tf_openai.py [hunk 4]
```diff
@@ -633,7 +610,7 @@ def call(
             mc_token_ids = inputs[6] if len(inputs) > 6 else mc_token_ids
             output_attentions = inputs[7] if len(inputs) > 7 else output_attentions
             assert len(inputs) <= 8, "Too many inputs."
-        elif isinstance(inputs, dict):
+        elif isinstance(inputs, (dict, BatchEncoding)):
```
Comment (LysandreJik) [inline src/transformers/modeling_tf_openai.py:613]: Non-cosmetic change

File: src/transformers/modeling_tf_roberta.py [hunk 5]
```diff
@@ -441,7 +414,7 @@ class TFRobertaForMultipleChoice(TFRobertaPreTrainedModel, TFMultipleChoiceLoss)
     def __init__(self, config, *inputs, **kwargs):
         super().__init__(config, *inputs, **kwargs)
 
-        self.roberta = TFBertMainLayer(config, name="roberta")
+        self.roberta = TFRobertaMainLayer(config, name="roberta")
```
Comment (LysandreJik) [inline src/transformers/modeling_tf_roberta.py:417]: Non cosmetic change

File: src/transformers/modeling_tf_transfo_xl.py [hunk 6]
```diff
@@ -863,7 +843,7 @@ def call(
             labels = inputs[4] if len(inputs) > 4 else labels
             output_attentions = inputs[5] if len(inputs) > 5 else output_attentions
             assert len(inputs) <= 6, "Too many inputs."
-        elif isinstance(inputs, dict):
+        elif isinstance(inputs, (BatchEncoding, dict)):
```
Comment (LysandreJik) [inline src/transformers/modeling_tf_transfo_xl.py:846]: Non cosmetic change

File: src/transformers/modeling_tf_xlm.py [hunk 7]
```diff
@@ -932,7 +890,9 @@ def call(
             head_mask = inputs[7] if len(inputs) > 7 else head_mask
             inputs_embeds = inputs[8] if len(inputs) > 8 else inputs_embeds
             output_attentions = inputs[9] if len(inputs) > 9 else output_attentions
-            assert len(inputs) <= 10, "Too many inputs."
+            output_hidden_states = inputs[10] if len(inputs) > 10 else output_hidden_states
+            labels = inputs[11] if len(inputs) > 11 else labels
+            assert len(inputs) <= 11, "Too many inputs."
```
Comment (LysandreJik) [inline src/transformers/modeling_tf_xlm.py:895]: Non cosmetic change

File: src/transformers/modeling_tf_xlm.py [hunk 8]
```diff
@@ -944,7 +904,9 @@ def call(
             head_mask = inputs.get("head_mask", head_mask)
             inputs_embeds = inputs.get("inputs_embeds", inputs_embeds)
             output_attentions = inputs.get("output_attentions", output_attentions)
-            assert len(inputs) <= 10, "Too many inputs."
+            output_hidden_states = inputs.get("output_hidden_states", output_hidden_states)
+            labels = inputs.get("labels", labels)
+            assert len(inputs) <= 12, "Too many inputs."
```
Comment (LysandreJik) [inline src/transformers/modeling_tf_xlm.py:909]: Non cosmetic change

File: src/transformers/modeling_tf_xlm.py [hunk 9]
```diff
@@ -1001,13 +963,14 @@ def __init__(self, config, *inputs, **kwargs):
         self.transformer = TFXLMMainLayer(config, name="transformer")
         self.dropout = tf.keras.layers.Dropout(config.dropout)
         self.classifier = tf.keras.layers.Dense(
-            config.num_labels, kernel_initializer=get_initializer(config.initializer_range), name="classifier"
+            config.num_labels, kernel_initializer=get_initializer(config.init_std), name="classifier"
```
Comment (LysandreJik) [inline src/transformers/modeling_tf_xlm.py:966]: Non cosmetic change

File: src/transformers/modeling_tf_xlm.py [hunk 10]
```diff
@@ -1041,22 +1004,16 @@ def call(
...
-        loss, scores = outputs[:2]
-
         """
+        if isinstance(inputs, (tuple, list)):
+            labels = inputs[11] if len(inputs) > 11 else labels
+            if len(inputs) > 11:
+                inputs = inputs[:11]
+        elif isinstance(inputs, (dict, BatchEncoding)):
+            labels = inputs.pop("labels", labels)
+
```
Comment (LysandreJik) [inline src/transformers/modeling_tf_xlm.py:1014]: Non cosmetic change

File: src/transformers/modeling_tf_xlm.py [hunk 11]
```diff
@@ -1072,7 +1029,7 @@ def call(
         sequence_output = self.dropout(sequence_output, training=training)
         logits = self.classifier(sequence_output)
 
-        outputs = (logits,) + transformer_outputs[2:]  # add hidden states and attention if they are here
+        outputs = (logits,) + transformer_outputs[1:]  # add hidden states and attention if they are here
```
Comment (LysandreJik) [inline src/transformers/modeling_tf_xlm.py:1032]: Non cosmetic change

File: docs/source/quicktour.rst [hunk 12]
```diff
@@ -1,378 +1,392 @@
...
+.. code-block::
+
+    >>> ## PYTORCH CODE
+    >>> model_name = "nlptown/bert-base-multilingual-uncased-sentiment"
+    >>> model = AutoModelForSequenceClassification.from_pretrained(model_name)
+    >>> tokenizer = AutoTokenizer.from_pretrained(model_name)
+    >>> pipe = pipeline('sentiment-analysis', model=model, tokenizer=tokenizer)
+    >>> ## TENSORFLOW CODE
+    >>> model_name = "nlptown/bert-base-multilingual-uncased-sentiment"
+    >>> model = TFAutoModelForSequenceClassification.from_pretrained(model_name, from_pt=True)  # This model only exists in PyTorch, so we use the `from_pt` flag to import that model in TensorFlow.
```
Comment (sgugger) [inline docs/source/quicktour.rst:114]:
```suggestion
    >>> # This model only exists in PyTorch, so we use the `from_pt` flag to import that model in TensorFlow.
    >>> model = TFAutoModelForSequenceClassification.from_pretrained(model_name, from_pt=True) 
```

File: src/transformers/modeling_tf_mobilebert.py [hunk 13]
```diff
@@ -926,9 +921,9 @@ def call(self, inputs, **kwargs):
         import tensorflow as tf
         from transformers import MobileBertTokenizer, TFMobileBertForPreTraining
 
-        tokenizer = MobileBertTokenizer.from_pretrained('mobilebert-uncased')
-        model = TFMobileBertForPreTraining.from_pretrained('mobilebert-uncased')
-        input_ids = tf.constant(tokenizer.encode("Hello, my dog is cute", add_special_tokens=True))[None, :]  # Batch size 1
+        tokenizer = MobileBertTokenizer.from_pretrained('google/mobilebert-uncased')
```
Comment (sgugger) [inline src/transformers/modeling_tf_mobilebert.py:924]: Those are not doctest, on purpose?

File: src/transformers/modeling_tf_mobilebert.py [hunk 14]
```diff
@@ -1041,12 +1025,12 @@ def call(self, inputs, **kwargs):
         import tensorflow as tf
         from transformers import MobileBertTokenizer, TFMobileBertForNextSentencePrediction
 
-        tokenizer = MobileBertTokenizer.from_pretrained('mobilebert-uncased')
-        model = TFMobileBertForNextSentencePrediction.from_pretrained('mobilebert-uncased')
+        tokenizer = MobileBertTokenizer.from_pretrained('google/mobilebert-uncased')
```
Comment (sgugger) [inline src/transformers/modeling_tf_mobilebert.py:1028]: Same here

File: src/transformers/modeling_tf_openai.py [hunk 15]
```diff
@@ -601,26 +581,23 @@ def call(
...
-        raise NotImplementedError
         tokenizer.add_special_tokens({'cls_token': '[CLS]'})
         model.resize_token_embeddings(len(tokenizer))  # Update the model embeddings with the new vocabulary size
         print(tokenizer.cls_token_id, len(tokenizer))  # The newly token the last token of the vocabulary
 
         choices = ["Hello, my dog is cute [CLS]", "Hello, my cat is cute [CLS]"]
-        input_ids = tf.constant([tokenizer.encode(s) for s in choices])[None, :]  # Batch size 1, 2 choices
-        mc_token_ids = tf.constant([input_ids.size(-1), input_ids.size(-1)])[None, :]  # Batch size 1
-        outputs = model(input_ids, mc_token_ids=mc_token_ids)
+        encoding = tokenizer(choices, return_tensors="tf")
```
Comment (sgugger) [inline src/transformers/modeling_tf_openai.py:596]: Same here

File: src/transformers/modeling_utils.py [hunk 16]
```diff
@@ -978,36 +978,38 @@ def generate(
 
         Examples::
 
+            from transformers import AutoTokenizer, AutoModelForCausalLM
```
Comment (sgugger) [inline src/transformers/modeling_utils.py:981]: Same here

File: src/transformers/modeling_xlm.py [hunk 17]
```diff
@@ -1006,19 +961,20 @@ def forward(
             Attentions weights after the attention softmax, used to compute the weighted average in the self-attention
             heads.
 
-    Examples::
+    Example::
```
Comment (sgugger) [inline src/transformers/modeling_xlm.py:964]: Same in this example

**Review by sgugger — APPROVED** [review]
Amazing work!
I just highlighted the examples I saw without doctest syntax, just to make sure it was on purpose.
Also there is one nit in quicktour.rst

**Review by thomwolf — APPROVED** [review]
Huge work!
Amazing!

File: src/transformers/modeling_utils.py (hunk 16 above)
Comment (LysandreJik) [reply src/transformers/modeling_utils.py:981]: Can't manage to have a clean doctest with such a big example

**Pull request merged**